backend/api/EventHandlers/InspectionFindingEventHandler.cs

```diff
@@ -23,12 +29,150 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 
                     logger.LogInformation("Found {count} inspection findings in the last {interval}.", inspectionFindings.Count, _timeSpan);
 
+                    if (inspectionFindings.Count > 0)
+                    {
+                        var findingsList = await GenerateFindingsList(inspectionFindings);
+
+                        string messageString = GenerateReportFromFindingsReportsList(findingsList);
+
+                        string adaptiveCardJson = GenerateAdaptiveCard(messageString);
+
+                        string url = GetWebhookURL("TeamsInspectionFindingsWebhook");
+
+                        var client = new HttpClient();
+                        client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                        var content = new StringContent(adaptiveCardJson, Encoding.UTF8, "application/json");
+                        var response = await client.PostAsync(url, content, stoppingToken);
+                        //if (response.StatusCode == StatusCodes.Status201Created) ;
+                    }
                 }
-                catch (OperationCanceledException)
+                catch (OperationCanceledException) { throw; }
+            }
+        }
+        private async Task<List<Finding>> GenerateFindingsList(List<InspectionFinding> inspectionFindings)
+        {
+            var findingsList = new List<Finding>();
+
+            foreach (var inspectionFinding in inspectionFindings)
+            {
+                try
                 {
-                    throw;
+                    var missionRun = await InspectionFindingService.GetMissionRunByIsarStepId(inspectionFinding);
+                    var task = await InspectionFindingService.GetMissionTaskByIsarStepId(inspectionFinding);
+
+                    if (task?.TagId != null && missionRun?.Area?.Plant?.Name != null && missionRun?.Area?.Name != null)
+                    {
+                        var finding = new Finding(
+                            task.TagId,
+                            missionRun.Area.Plant.Name,
+                            missionRun.Area.Name,
+                            inspectionFinding.Finding,
+                            inspectionFinding.InspectionDate,
+                            missionRun.Robot.Name
+                        );
+
+                        findingsList.Add(finding);
+                    }
+                    else
+                    {
+                        logger.LogInformation("Failed to generate a finding since TagId in missionTask or Area in MissionRun is null");
+                        continue;
+                    }
                 }
+                catch { throw; }
```

Comment:
Again, I'm not sure if this does anything. If you want the exception to stil be passed on then you could maybe log something here. Otherwise you don't need the catch

Reply:
It could be good to log the error and then return, to make the logs cleaner

Reply:
Yup agreed.